Userland/cal.cpp

```diff
@@ -98,6 +98,70 @@ void clean_buffers()
     temp_buffer[line_width - 1] = '\0';
 }
 
+void append_to_print(char* buffer, int row, int column, char* text)
+{
+    int starting_point = (line_width * row) + (column * column_width);
+    for (int i = 0; text[i] != '\0'; i++) {
+        buffer[starting_point + i] = text[i];
+    }
+}
+
+void insert_month_to_print(int column, int month, int year)
+{
+    int printing_column = column;
+    int printing_row = 0;
+
+    // FIXME: Both the month name and month header text should be provided by a locale
+    sprintf(temp_buffer, "     %02u - %04u    ", month, year);
+    append_to_print(print_buffer, printing_row, printing_column, temp_buffer);
+    printing_row++;
+
+    sprintf(temp_buffer, "Su Mo Tu We Th Fr Sa");
+    append_to_print(print_buffer, printing_row, printing_column, temp_buffer);
+    printing_row++;
+
+    int day_to_print = 1;
+    int first_day_of_week_for_month = day_of_week(1, month, year);
+    int days_in_the_month = get_number_of_days(month, year);
+    int last_written_chars = 0;
+    for (int i = 1; day_to_print <= days_in_the_month; ++i) {
+        if (i - 1 < first_day_of_week_for_month) {
+            last_written_chars += sprintf(temp_buffer + last_written_chars * sizeof(char), "   ");
```

Comment:
It's unidiomatic to write `sizeof(char)`, because it's always guaranteed to be exactly 1.

Reply:
You're right. Something that I learnt today :)